[Samyukta] iP

MontyPython28 opened the pull request:
# Ace Attorney Bot
> “While others were dreaming about it, I was getting it done.” ― Nathan W. Morris  ([source](https://www.goodreads.com/quotes/tag/getting-things-done))

## OBJECTION!
With the Ace Attorney Bot, you don't need to carry all the things you need to do in your head. It's

- text-based
- easy to learn
- ~~FAST~~ _SUPER FAST_ to use

You just need to do the following steps:

1. Download the aceacttorney.jar file from [here](https://github.com/MontyPython28/ip/releases/tag/A-Release)
2. Double-click it
3. Add your tasks
4. Let it manage its tasks for you 

And... it's **FREE!!!** 🥳 🥳 🥳 

Features:

- [x] Managing Tasks
- [x] Prioritizing Tasks
- [ ]  Reminders (coming soon)

If you want to practice Java, you're welcome to look at the source code! The `main` method in the `Launcher` class starts the application.

Commit:
Add Gradle support

MontyPython28 committed:
Level-1: Greet, Echo, Exit

MontyPython28 committed:
Level-2: Add, List

MontyPython28 committed:
Level-3: Mark as Done

MontyPython28 committed:
Level-4: ToDos, Events, Deadlines

MontyPython28 committed:
A-TextUiTesting

MontyPython28 committed:
Level-5: Handle Errors

MontyPython28 committed:
Level-6: Delete

MontyPython28 committed:
Created classes for Inheritance; added toString() methods; fixed problems with deadline and event

MontyPython28 committed:
Made Task abstract; made DukeException class; fixed and updated files for A-TextUiTesting.

MontyPython28 committed:
Level-7: Save

MontyPython28 committed:
Level-8: Dates and Times

MontyPython28 committed:
Merge pull request #1 from MontyPython28/branch-Level-7

Level-7: Save

MontyPython28 committed:
Merge branch 'master' into branch-Level-8

MontyPython28 committed:
Merge pull request #2 from MontyPython28/branch-Level-8

Level-8: Dates and Times

MontyPython28 committed:
Intermediate state of converting to more OOP state

MontyPython28 committed:
A-MoreOOP

MontyPython28 committed:
A-Packages; also added Exceptions in Storage, Parser and Commands classes

MontyPython28 committed:
Unit tests: add tests for Parser and TaskList

MontyPython28 committed:
Add FindCommand functionality

MontyPython28 committed:
Add JavaDocs to classes

MontyPython28 committed:
Standardise to coding convention

MontyPython28 committed:
Merge pull request #3 from MontyPython28/branch-Level-9

Add FindCommand functionality

MontyPython28 committed:
Merge pull request #4 from MontyPython28/branch-A-JavaDoc

Add JavaDocs to classes

MontyPython28 committed:
Merge branch 'master' into branch-A-CodingStandard

MontyPython28 committed:
Merge pull request #5 from MontyPython28/branch-A-CodingStandard

Standardise to coding convention

MontyPython28 committed:
Cleaning up and fixing errors

MontyPython28 committed:
Merge pull request #6 from nus-cs2103-AY2122S2/add-gradle-support

Add Gradle support files

MontyPython28 committed:
Merge pull request #7 from MontyPython28/branch-A-Gradle

Branch A-Gradle

MontyPython28 committed:
Add Checkstyle config files and made appropriate changes to follow Java coding standards

MontyPython28 committed:
Merge pull request #8 from MontyPython28/branch-A-CheckStyle

Add Checkstyle config files and made appropriate changes to follow Ja…

yl-ang commented on src/main/java/Parser.java:
```diff
@@ -0,0 +1,87 @@
+import command.TodoCommand;
+import command.UnmarkCommand;
+import exception.DukeException;
+
```
perhaps can add class header comment here

yl-ang commented on src/main/java/exception/DukeException.java:
```diff
@@ -0,0 +1,12 @@
+
+    /** Constructor helps format error message. */
+    public DukeException(String errorMessage) {
+        super("____________________________________________________________\n"
```
perhaps can let Ui.java handle the line formatting to recycle code

MontyPython28 replied:
Thank you very much for all your comments! I've made the changes you suggested- definitely very helpful

yl-ang reviewed (commented):
Overall, good work with maintaining code quality. Good structuring of code, good avoidance of long methods, and deep nesting.

MontyPython28 committed:
Add Java GUI to iP (Level-10)

MontyPython28 committed:
Merge pull request #9 from MontyPython28/branch-Level-10

Add Java GUI to iP (Level-10)

MontyPython28 committed:
Add assertions to classes

We want to make sure that our file handling and exception handling work as predicted.

Let's-
* add an assertion to check that the new file's path and filename are in the correct format
* add assertions to DeleteCommand, MarkCommand and UnmarkCommand to check the post-condition of our exception handling

MontyPython28 committed:
Merge pull request #10 from MontyPython28/branch-A-Assertions

Branch A-Assertions

MontyPython28 committed:
Improve code quality

We want to make sure that our code is more readable.

Most work has been done on the Parser file.

MontyPython28 committed:
Merge pull request #11 from MontyPython28/branch-A-CodeQuality

Improve code quality

MontyPython28 committed:
Implement C-Priority

We want to allow users to optionally add a Priority to their tasks. This should be reflected in the TaskList and the Storage.

MontyPython28 committed:
Merge pull request #12 from MontyPython28/branch-C-Priority

Implement C-Priority

MontyPython28 committed:
Implement better GUI

MontyPython28 committed:
Add User Guide and contains code acknowledgements

MontyPython28 committed:
Add Ui.png

MontyPython28 committed:
Set theme jekyll-theme-merlot

MontyPython28 committed:
Fix acknowledgement

MontyPython28 committed:
Merge remote-tracking branch 'origin/master'

MontyPython28 committed:
Edit ListCommand

MontyPython28 committed:
Fix Parser

MontyPython28 committed:
Change Task toString methods

MontyPython28 committed:
Update README.md in docs
